packages/react-dom-bindings/src/client/ReactDOMHostConfig.js

```diff
@@ -1389,3 +1564,22 @@ export {
   acquireResource,
   releaseResource,
 } from './ReactDOMFloatClient';
+
+// -------------------
+//     Singletons
+// -------------------
+
+export const supportsSingletons = true;
+
+export function isHostSingletonInstance(
+  instance: Instance | Container,
+): boolean {
+  if (instance.nodeType === ELEMENT_NODE) {
+    const element: Element = (instance: any);
+    return isHostSingletonType(element.tagName.toLowerCase());
```

Comment:
I noticed that because of this second usage, this function doesn't get inlined into the reconciler. We shouldn't use this again here. In fact, we probably shouldn't even have this `isHostSingletonInstance` file. Just inline all of it into getNextHydratable.

Reply:
rewrote and eliminated this function